---
title: "RNA-seq Data Analysis"
author: "Jialin Ma"
date: "October 22, 2018"
output:
  html_document:
    self_contained: false
    toc: true
  pdf_document: default
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, collapse = FALSE)
```

## Load the packages

```{r}
suppressPackageStartupMessages({
    library(here)
    library(dplyr)
    library(ggplot2)
    library(pheatmap)
    library(RColorBrewer)
    library(EnsDb.Hsapiens.v86)
    library(DESeq2)
})
```

## Load the dataset

The transcript quantification results have been produced by `salmon`.
The `tximport` package can import and summarize the information on gene level.

```{r}
library(tximport)
tx2gene <- transcripts(EnsDb.Hsapiens.v86,
        return.type = "data.frame", columns = c("tx_id", "gene_id"))
salmon_dirs <- list.files(here("result/salmon_quant"), full.names = TRUE)
salmon_files <- sapply(salmon_dirs, function(x) file.path(x, "quant.sf"))
names(salmon_files) <- basename(salmon_dirs)
txi <- tximport(salmon_files, type = "salmon", tx2gene = tx2gene, ignoreTxVersion = TRUE)
names(txi)
```

We will use `DESeq2` for further analysis.
Convert the data to DESeqDataSet class.

```{r}
samples <- data.frame(condition = c(rep.int("control", 3), rep.int("treat", 4)),
                      sample_name = names(salmon_files))
rownames(samples) <- names(salmon_files)
samples
dds <- DESeqDataSetFromTximport(txi, colData = samples, design = ~ condition)
dds
```

Remove the empty rows.

```{r}
nrow(dds)
dds <- dds[ rowSums(counts(dds)) > 0, ]
nrow(dds)
```



## Regularized-logarithm transformation (rlog)

We will use the regularized log transformation for PCA plot and SD-mean plot.
It will reduce the variation on lowly expressed genes.

```{r}
rld <- rlog(dds, blind = FALSE)
head(assay(rld), 3)
```

## Heatmap showing sample distance (rlog transformed)

The samples can be clearly clustered by the control/treat condition.

```{r}
sampleDists <- dist(t(assay(rld)))
sampleDistMatrix <- as.matrix( sampleDists )
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors)
```

## PCA plot (rlog transformed)

Previously we have conducted PCA analysis on samples. Here do PCA plot on genes.

```{r}
plotPCA(rld)
```

## Differential expression analysis

Run the differential expression pipeline.

```{r}
dds <- DESeq(dds)
```

Then build the results table, we use an adjusted p-value cutoff by 0.05.

```{r}
res <- results(dds, alpha = 0.05)
head(res[order(res$pvalue), ], n = 3)
mcols(res, use.names = TRUE)
```

```{r}
summary(res)
```


```{r eval=FALSE, include=FALSE}
## Log fold change shrinkage for visualization and ranking
resLFC <- lfcShrink(dds, coef="condition_treat_vs_control", type="apeglm")
resLFC
```

## MA plot

Points are be colored red if the adjusted p value is less than 0.05.

```{r}
plotMA(res)
```

## Distribution of p values

```{r}
hist(res$pvalue[res$baseMean > 1], breaks = 0:20/20,
     col = "grey50", border = "white")
```


## Annotating the gene names

```{r}
library(AnnotationDbi)
library(org.Hs.eg.db)
res$symbol <- mapIds(org.Hs.eg.db, keys=row.names(res),
    column="SYMBOL", keytype="ENSEMBL", multiVals="first")
res <- res[, c(ncol(res), 1:(ncol(res)-1))]
head(res[order(res$pvalue),], n = 3)
```

## Generate a table of differentially expressed genes

We will select the genes that have adjusted p value < 0.05 and order
them by their fold change.

Also, since there are a lot of genes differentially expressed, we may be
interested in ones that have larger fold change. We will select the ones
that have abs(log2foldchange) > 2.


```{r}
selected <- res[!is.na(res$padj), ]
selected <- selected[selected$padj < 0.05, ]
```

We selected `r nrow(selected)` out of `r nrow(res)` genes in the step above.

The following is the distribution of log2foldchange in genes with
adjusted p value < 0.05. The yellow bars are ones with fold change < 2, which
will be filtered.

```{r}
hist(sort(selected$log2FoldChange), breaks = 30,
     col = c(rep("white", 14), rep("yellow", 2), rep("white", 14)), xlim = c(-15, 15))
selected <- selected[abs(selected$log2FoldChange) > 2, ]
selected <- selected[order(abs(selected$log2FoldChange), decreasing = TRUE), ]
```

After filtering by fold change, we finally selected `r nrow(selected)` genes.

Here we generate a table showing the top 100 rows.

```{r}
selected <- as.data.frame(selected)
selected$log10pvalue <- log10(selected$pvalue)
selected$log10padj <- log10(selected$padj)
selected <- selected[ , c("symbol", "baseMean", "log2FoldChange", "pvalue",
                         "log10pvalue", "padj", "log10padj")]
```

```{r}
knitr::kable(selected[1:100,])
```


## Gene clustering

### Clustering of all selected genes by expression level (rlog transformed)

```{r}
library(genefilter)
mat <- assay(rld)[rownames(selected), ]
anno <- as.data.frame(colData(rld)[, c("condition", "sample_name")])
pheatmap(mat, annotation_col = anno, show_rownames = FALSE,
         main = "Heatmap of rlog-transformed gene expression levels")
```

### Clustering of all selected genes by relative expression levels across samples

Instead of absolute expression strength, the following plot uses the amount by which
each gene deviates in a specific sample from the gene's average across all samples.

The trees on top shows the clustering by samples, the trees on left shows the clustering
by genes. It can be shown that the control and treat samples are clearly separated and
the genes may be roughly divided into two clusters (up-regulated and down-regulated).

```{r}
mat <- mat - rowMeans(mat)
pheatmap(mat, annotation_col = anno, show_rownames = FALSE,
         main = "Heatmap of relative rlog-transformed values across samples")
```

### Clustering of genes with the highest variance across samples

Here we select the 30 genes with the highest variance of expression values across
samples.

```{r}
library(genefilter)
topVarGenes <- head(order(rowVars(assay(rld)), decreasing = TRUE), 30)
mat <- assay(rld)[topVarGenes, ]
mat <- mat[rownames(mat) %in% rownames(selected), ]
mat <- mat - rowMeans(mat)
anno <- as.data.frame(colData(rld)[, c("condition", "sample_name")])
pheatmap(mat, annotation_col = anno,
         main = "Heatmap of relative rlog-transformed values across samples")
```


## PCA plot of genes

We will use rlog-transformed values. The following is a PCA plot of all
selected genes.

```{r}
library(ggfortify)
autoplot(prcomp(assay(rld)[rownames(assay(rld)) %in% rownames(selected), ]), size = 1, alpha = 0.8)
```

It is clear that the genes can be separated by two parts by the second dimension.
If we plot all the genes like the following,
it will be clear for us that the second dimension aligns with the foldchange -- since
we have filtered the genes with low foldchange. Thus the two clusters are
up-regulated genes and down-regulated genes.

```{r}
autoplot(prcomp(assay(rld)[rownames(assay(rld)), ]), size = 1, alpha = 0.8)
```

### PCA plot of genes with high variance

Again, we select the genes with the highest variance across samples:

```{r}
topVarGenes <- head(order(rowVars(assay(rld)), decreasing = TRUE), 50)
tmp <- assay(rld)[topVarGenes, ]
tmp <- tmp[rownames(tmp) %in% rownames(selected), ]
autoplot(prcomp(tmp))
```


## Pathway analysis

The package `ReactomePA` offers the possibility to test enrichment of specific
pathways using the free, open-source, curated and peer reviewed pathway Reactome
pathway database.

```{r}
library(ReactomePA)
entrez_ids <- mapIds(org.Hs.eg.db, keys = rownames(res),
                     column = "ENTREZID", keytype = "ENSEMBL", multiVals = "first")
entrez_ids <- entrez_ids[rownames(selected)]
entrez_ids <- entrez_ids[!is.na(entrez_ids)]

reactome_enrich <- enrichPathway(gene = entrez_ids, organism = "human", readable = TRUE)
```

Then we visualize the enriched terms.

```{r fig.width=12, fig.height=10}
barplot(reactome_enrich, showCategory = 25)
dotplot(reactome_enrich, showCategory = 25)
```

```{r fig.width=12, fig.height=10}
ReactomePA::emapplot(reactome_enrich, showCategory = Inf)
```

### Pathway analysis of up-regulated genes

```{r fig.width=12, fig.height=10}
up_regulated <- selected[selected$log2FoldChange > 0, ]
up_regulated <- up_regulated[rownames(up_regulated) %in% names(entrez_ids), ]
tryCatch(
    ReactomePA::emapplot(
        enrichPathway(gene = entrez_ids[rownames(up_regulated)], organism = "human"),
        showCategory = Inf),
    error = function(e) print(e$message)
)
```

However we do not have any term enriched given the up-regulated genes.

### Pathway analysis of down-regulated genes

```{r fig.width=12, fig.height=10}
down_regulated <- selected[selected$log2FoldChange < 0, ]
down_regulated <- down_regulated[rownames(down_regulated) %in% names(entrez_ids), ]
ReactomePA::emapplot(
    enrichPathway(gene = entrez_ids[rownames(down_regulated)], organism = "human"),
    showCategory = Inf)
```



