<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jialin Ma" />

<meta name="date" content="2018-10-22" />

<title>RNA-seq Data Analysis</title>

<script src="rnaseq_explore_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="rnaseq_explore_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="rnaseq_explore_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="rnaseq_explore_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="rnaseq_explore_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="rnaseq_explore_files/navigation-1.1/tabsets.js"></script>
<link href="rnaseq_explore_files/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="rnaseq_explore_files/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="fluid-row" id="header">



<h1 class="title toc-ignore">RNA-seq Data Analysis</h1>
<h4 class="author"><em>Jialin Ma</em></h4>
<h4 class="date"><em>October 22, 2018</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#load-the-packages">Load the packages</a></li>
<li><a href="#load-the-dataset">Load the dataset</a></li>
<li><a href="#regularized-logarithm-transformation-rlog">Regularized-logarithm transformation (rlog)</a></li>
<li><a href="#heatmap-showing-sample-distance-rlog-transformed">Heatmap showing sample distance (rlog transformed)</a></li>
<li><a href="#pca-plot-rlog-transformed">PCA plot (rlog transformed)</a></li>
<li><a href="#differential-expression-analysis">Differential expression analysis</a></li>
<li><a href="#ma-plot">MA plot</a></li>
<li><a href="#distribution-of-p-values">Distribution of p values</a></li>
<li><a href="#annotating-the-gene-names">Annotating the gene names</a></li>
<li><a href="#generate-a-table-of-differentially-expressed-genes">Generate a table of differentially expressed genes</a></li>
<li><a href="#gene-clustering">Gene clustering</a><ul>
<li><a href="#clustering-of-all-selected-genes-by-expression-level-rlog-transformed">Clustering of all selected genes by expression level (rlog transformed)</a></li>
<li><a href="#clustering-of-all-selected-genes-by-relative-expression-levels-across-samples">Clustering of all selected genes by relative expression levels across samples</a></li>
<li><a href="#clustering-of-genes-with-the-highest-variance-across-samples">Clustering of genes with the highest variance across samples</a></li>
</ul></li>
<li><a href="#pca-plot-of-genes">PCA plot of genes</a><ul>
<li><a href="#pca-plot-of-genes-with-high-variance">PCA plot of genes with high variance</a></li>
</ul></li>
</ul>
</div>

<div id="load-the-packages" class="section level2">
<h2>Load the packages</h2>
<pre class="r"><code>suppressPackageStartupMessages({
    library(here)
    library(dplyr)
    library(ggplot2)
    library(pheatmap)
    library(RColorBrewer)
    library(EnsDb.Hsapiens.v86)
    library(DESeq2)
})</code></pre>
</div>
<div id="load-the-dataset" class="section level2">
<h2>Load the dataset</h2>
<p>The transcript quantification results have been produced by <code>salmon</code>. The <code>tximport</code> package can import and summarize the information on gene level.</p>
<pre class="r"><code>library(tximport)
tx2gene &lt;- transcripts(EnsDb.Hsapiens.v86,
        return.type = &quot;data.frame&quot;, columns = c(&quot;tx_id&quot;, &quot;gene_id&quot;))
salmon_dirs &lt;- list.files(here(&quot;result/salmon_quant&quot;), full.names = TRUE)
salmon_files &lt;- sapply(salmon_dirs, function(x) file.path(x, &quot;quant.sf&quot;))
names(salmon_files) &lt;- basename(salmon_dirs)
txi &lt;- tximport(salmon_files, type = &quot;salmon&quot;, tx2gene = tx2gene, ignoreTxVersion = TRUE)</code></pre>
<pre><code>## reading in files with read_tsv</code></pre>
<pre><code>## 1 2 3 4 5 6 7 
## transcripts missing from tx2gene: 7499
## summarizing abundance
## summarizing counts
## summarizing length</code></pre>
<pre class="r"><code>names(txi)</code></pre>
<pre><code>## [1] &quot;abundance&quot;           &quot;counts&quot;              &quot;length&quot;             
## [4] &quot;countsFromAbundance&quot;</code></pre>
<p>We will use <code>DESeq2</code> for further analysis. Convert the data to DESeqDataSet class.</p>
<pre class="r"><code>samples &lt;- data.frame(condition = c(rep.int(&quot;control&quot;, 3), rep.int(&quot;treat&quot;, 4)),
                      sample_name = names(salmon_files))
rownames(samples) &lt;- names(salmon_files)
samples</code></pre>
<pre><code>##              condition  sample_name
## control_rep1   control control_rep1
## control_rep2   control control_rep2
## control_rep3   control control_rep3
## treat_rep1       treat   treat_rep1
## treat_rep2       treat   treat_rep2
## treat_rep3       treat   treat_rep3
## treat_rep4       treat   treat_rep4</code></pre>
<pre class="r"><code>dds &lt;- DESeqDataSetFromTximport(txi, colData = samples, design = ~ condition)</code></pre>
<pre><code>## using counts and average transcript lengths from tximport</code></pre>
<pre class="r"><code>dds</code></pre>
<pre><code>## class: DESeqDataSet 
## dim: 37361 7 
## metadata(1): version
## assays(2): counts avgTxLength
## rownames(37361): ENSG00000000003 ENSG00000000005 ...
##   ENSG00000283698 ENSG00000283700
## rowData names(0):
## colnames(7): control_rep1 control_rep2 ... treat_rep3 treat_rep4
## colData names(2): condition sample_name</code></pre>
<p>Remove the empty rows.</p>
<pre class="r"><code>nrow(dds)</code></pre>
<pre><code>## [1] 37361</code></pre>
<pre class="r"><code>dds &lt;- dds[ rowSums(counts(dds)) &gt; 0, ]
nrow(dds)</code></pre>
<pre><code>## [1] 25001</code></pre>
</div>
<div id="regularized-logarithm-transformation-rlog" class="section level2">
<h2>Regularized-logarithm transformation (rlog)</h2>
<p>We will use the regularized log transformation for PCA plot and SD-mean plot. It will reduce the variation on lowly expressed genes.</p>
<pre class="r"><code>rld &lt;- rlog(dds, blind = FALSE)</code></pre>
<pre><code>## using &#39;avgTxLength&#39; from assays(dds), correcting for library size</code></pre>
<pre class="r"><code>head(assay(rld), 3)</code></pre>
<pre><code>##                 control_rep1 control_rep2 control_rep3 treat_rep1
## ENSG00000000003    11.300466    11.231395    11.104771  12.783082
## ENSG00000000419    11.171577    11.132015    11.321105  11.167904
## ENSG00000000457     8.207076     8.139017     7.899893   8.840509
##                 treat_rep2 treat_rep3 treat_rep4
## ENSG00000000003  12.758421  12.810201  12.789334
## ENSG00000000419  10.826120  10.920410  11.130422
## ENSG00000000457   8.943653   8.884305   8.846911</code></pre>
</div>
<div id="heatmap-showing-sample-distance-rlog-transformed" class="section level2">
<h2>Heatmap showing sample distance (rlog transformed)</h2>
<p>The samples can be clearly clustered by the control/treat condition.</p>
<pre class="r"><code>sampleDists &lt;- dist(t(assay(rld)))
sampleDistMatrix &lt;- as.matrix( sampleDists )
colnames(sampleDistMatrix) &lt;- NULL
colors &lt;- colorRampPalette( rev(brewer.pal(9, &quot;Blues&quot;)) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors)</code></pre>
<p><img src="rnaseq_explore_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
</div>
<div id="pca-plot-rlog-transformed" class="section level2">
<h2>PCA plot (rlog transformed)</h2>
<p>Previously we have conducted PCA analysis on samples. Here do PCA plot on genes.</p>
<pre class="r"><code>plotPCA(rld)</code></pre>
<p><img src="rnaseq_explore_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
</div>
<div id="differential-expression-analysis" class="section level2">
<h2>Differential expression analysis</h2>
<p>Run the differential expression pipeline.</p>
<pre class="r"><code>dds &lt;- DESeq(dds)</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## using &#39;avgTxLength&#39; from assays(dds), correcting for library size</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<p>Then build the results table, we use an adjusted p-value cutoff by 0.05.</p>
<pre class="r"><code>res &lt;- results(dds, alpha = 0.05)
head(res[order(res$pvalue), ], n = 3)</code></pre>
<pre><code>## log2 fold change (MLE): condition treat vs control 
## Wald test p-value: condition treat vs control 
## DataFrame with 3 rows and 6 columns
##                         baseMean    log2FoldChange             lfcSE
##                        &lt;numeric&gt;         &lt;numeric&gt;         &lt;numeric&gt;
## ENSG00000241343 10887.8871376054 -4.06463847625495 0.108594662387474
## ENSG00000185950 7290.71556646402  3.71238174924945 0.103648480871635
## ENSG00000183054 3308.40662655646  5.76746803913769 0.166493932393504
##                              stat                pvalue
##                         &lt;numeric&gt;             &lt;numeric&gt;
## ENSG00000241343 -37.4294499093518 1.29721089573246e-306
## ENSG00000185950  35.8170396520052 5.99696386772001e-281
## ENSG00000183054  34.6407100620727 6.16464222513363e-263
##                                  padj
##                             &lt;numeric&gt;
## ENSG00000241343  2.4815644435362e-302
## ENSG00000185950 5.73609593947419e-277
## ENSG00000183054 3.93098685889354e-259</code></pre>
<pre class="r"><code>mcols(res, use.names = TRUE)</code></pre>
<pre><code>## DataFrame with 6 rows and 2 columns
##                        type
##                 &lt;character&gt;
## baseMean       intermediate
## log2FoldChange      results
## lfcSE               results
## stat                results
## pvalue              results
## padj                results
##                                                       description
##                                                       &lt;character&gt;
## baseMean                mean of normalized counts for all samples
## log2FoldChange log2 fold change (MLE): condition treat vs control
## lfcSE                  standard error: condition treat vs control
## stat                   Wald statistic: condition treat vs control
## pvalue              Wald test p-value: condition treat vs control
## padj                                         BH adjusted p-values</code></pre>
<pre class="r"><code>summary(res)</code></pre>
<pre><code>## 
## out of 25001 with nonzero total read count
## adjusted p-value &lt; 0.05
## LFC &gt; 0 (up)       : 5536, 22%
## LFC &lt; 0 (down)     : 5371, 21%
## outliers [1]       : 54, 0.22%
## low counts [2]     : 5817, 23%
## (mean count &lt; 1)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
</div>
<div id="ma-plot" class="section level2">
<h2>MA plot</h2>
<p>Points are be colored red if the adjusted p value is less than 0.05.</p>
<pre class="r"><code>plotMA(res)</code></pre>
<p><img src="rnaseq_explore_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
</div>
<div id="distribution-of-p-values" class="section level2">
<h2>Distribution of p values</h2>
<pre class="r"><code>hist(res$pvalue[res$baseMean &gt; 1], breaks = 0:20/20,
     col = &quot;grey50&quot;, border = &quot;white&quot;)</code></pre>
<p><img src="rnaseq_explore_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
</div>
<div id="annotating-the-gene-names" class="section level2">
<h2>Annotating the gene names</h2>
<pre class="r"><code>library(AnnotationDbi)
library(org.Hs.eg.db)
res$symbol &lt;- mapIds(org.Hs.eg.db, keys=row.names(res),
    column=&quot;SYMBOL&quot;, keytype=&quot;ENSEMBL&quot;, multiVals=&quot;first&quot;)</code></pre>
<pre><code>## &#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre class="r"><code>res &lt;- res[, c(ncol(res), 1:(ncol(res)))]
head(res[order(res$pvalue),], n = 3)</code></pre>
<pre><code>## log2 fold change (MLE): condition treat vs control 
## Wald test p-value: condition treat vs control 
## DataFrame with 3 rows and 8 columns
##                      symbol         baseMean    log2FoldChange
##                 &lt;character&gt;        &lt;numeric&gt;         &lt;numeric&gt;
## ENSG00000241343      RPL36A 10887.8871376054 -4.06463847625495
## ENSG00000185950        IRS2 7290.71556646402  3.71238174924945
## ENSG00000183054       RGPD6 3308.40662655646  5.76746803913769
##                             lfcSE              stat                pvalue
##                         &lt;numeric&gt;         &lt;numeric&gt;             &lt;numeric&gt;
## ENSG00000241343 0.108594662387474 -37.4294499093518 1.29721089573246e-306
## ENSG00000185950 0.103648480871635  35.8170396520052 5.99696386772001e-281
## ENSG00000183054 0.166493932393504  34.6407100620727 6.16464222513363e-263
##                                  padj    symbol.1
##                             &lt;numeric&gt; &lt;character&gt;
## ENSG00000241343  2.4815644435362e-302      RPL36A
## ENSG00000185950 5.73609593947419e-277        IRS2
## ENSG00000183054 3.93098685889354e-259       RGPD6</code></pre>
</div>
<div id="generate-a-table-of-differentially-expressed-genes" class="section level2">
<h2>Generate a table of differentially expressed genes</h2>
<p>We will select the genes that have adjusted p value &lt; 0.05 and order them by their fold change.</p>
<p>Also, since there are a lot of genes differentially expressed, we may be interested in ones that have larger fold change. We will select the ones that have abs(log2foldchange) &gt; 2.</p>
<pre class="r"><code>selected &lt;- res[!is.na(res$padj), ]
selected &lt;- selected[selected$padj &lt; 0.05, ]</code></pre>
<p>The following is the distribution of log2foldchange in genes with adjusted p value &lt; 0.05. The yellow bars are ones with fold change &lt; 2, which will be filtered.</p>
<pre class="r"><code>hist(sort(selected$log2FoldChange), breaks = 30,
     col = c(rep(&quot;white&quot;, 14), rep(&quot;yellow&quot;, 2), rep(&quot;white&quot;, 14)), xlim = c(-15, 15))</code></pre>
<p><img src="rnaseq_explore_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<pre class="r"><code>selected &lt;- selected[abs(selected$log2FoldChange) &gt; 2, ]
selected &lt;- selected[order(abs(selected$log2FoldChange), decreasing = TRUE), ]</code></pre>
<pre class="r"><code>selected &lt;- as.data.frame(selected)
selected$log10pvalue &lt;- log10(selected$pvalue)
selected$log10padj &lt;- log10(selected$padj)
selected &lt;- selected[ , c(&quot;symbol&quot;, &quot;baseMean&quot;, &quot;log2FoldChange&quot;, &quot;pvalue&quot;,
                         &quot;log10pvalue&quot;, &quot;padj&quot;, &quot;log10padj&quot;)]</code></pre>
<p>Here we generate a table showing the top 100 rows.</p>
<pre class="r"><code>knitr::kable(selected[1:100,])</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="left">symbol</th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">log10pvalue</th>
<th align="right">padj</th>
<th align="right">log10padj</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSG00000235655</td>
<td align="left">NA</td>
<td align="right">3076.70095</td>
<td align="right">15.646846</td>
<td align="right">0.0000000</td>
<td align="right">-37.773375</td>
<td align="right">0.0000000</td>
<td align="right">-36.462471</td>
</tr>
<tr class="even">
<td>ENSG00000282230</td>
<td align="left">ADAM9</td>
<td align="right">2888.90683</td>
<td align="right">15.556043</td>
<td align="right">0.0000000</td>
<td align="right">-27.714462</td>
<td align="right">0.0000000</td>
<td align="right">-26.604766</td>
</tr>
<tr class="odd">
<td>ENSG00000244398</td>
<td align="left">NA</td>
<td align="right">1581.68738</td>
<td align="right">14.687067</td>
<td align="right">0.0000000</td>
<td align="right">-34.269128</td>
<td align="right">0.0000000</td>
<td align="right">-33.018822</td>
</tr>
<tr class="even">
<td>ENSG00000283485</td>
<td align="left">NA</td>
<td align="right">3148.47842</td>
<td align="right">-14.678216</td>
<td align="right">0.0000000</td>
<td align="right">-45.587733</td>
<td align="right">0.0000000</td>
<td align="right">-44.139802</td>
</tr>
<tr class="odd">
<td>ENSG00000280830</td>
<td align="left">ADI1</td>
<td align="right">1387.78347</td>
<td align="right">14.498522</td>
<td align="right">0.0000000</td>
<td align="right">-33.249811</td>
<td align="right">0.0000000</td>
<td align="right">-32.021942</td>
</tr>
<tr class="even">
<td>ENSG00000234851</td>
<td align="left">NA</td>
<td align="right">12044.69368</td>
<td align="right">-14.210605</td>
<td align="right">0.0000000</td>
<td align="right">-83.041873</td>
<td align="right">0.0000000</td>
<td align="right">-80.980266</td>
</tr>
<tr class="odd">
<td>ENSG00000235043</td>
<td align="left">NA</td>
<td align="right">802.98353</td>
<td align="right">13.709034</td>
<td align="right">0.0000000</td>
<td align="right">-23.414630</td>
<td align="right">0.0000000</td>
<td align="right">-22.418247</td>
</tr>
<tr class="even">
<td>ENSG00000204388</td>
<td align="left">HSPA1B</td>
<td align="right">736.70040</td>
<td align="right">13.584974</td>
<td align="right">0.0000000</td>
<td align="right">-29.621084</td>
<td align="right">0.0000000</td>
<td align="right">-28.466473</td>
</tr>
<tr class="odd">
<td>ENSG00000255072</td>
<td align="left">PIGY</td>
<td align="right">732.20170</td>
<td align="right">13.575567</td>
<td align="right">0.0000000</td>
<td align="right">-29.329213</td>
<td align="right">0.0000000</td>
<td align="right">-28.182312</td>
</tr>
<tr class="even">
<td>ENSG00000220842</td>
<td align="left">NA</td>
<td align="right">659.11785</td>
<td align="right">13.424178</td>
<td align="right">0.0000000</td>
<td align="right">-24.542919</td>
<td align="right">0.0000000</td>
<td align="right">-23.518162</td>
</tr>
<tr class="odd">
<td>ENSG00000213442</td>
<td align="left">NA</td>
<td align="right">428.76061</td>
<td align="right">12.804108</td>
<td align="right">0.0000000</td>
<td align="right">-25.512788</td>
<td align="right">0.0000000</td>
<td align="right">-24.461266</td>
</tr>
<tr class="even">
<td>ENSG00000283229</td>
<td align="left">NA</td>
<td align="right">426.25001</td>
<td align="right">12.795152</td>
<td align="right">0.0000346</td>
<td align="right">-4.460787</td>
<td align="right">0.0000914</td>
<td align="right">-4.038870</td>
</tr>
<tr class="odd">
<td>ENSG00000269378</td>
<td align="left">NA</td>
<td align="right">839.92889</td>
<td align="right">-12.771995</td>
<td align="right">0.0000000</td>
<td align="right">-34.703289</td>
<td align="right">0.0000000</td>
<td align="right">-33.441105</td>
</tr>
<tr class="even">
<td>ENSG00000204072</td>
<td align="left">NA</td>
<td align="right">405.48635</td>
<td align="right">12.723710</td>
<td align="right">0.0000000</td>
<td align="right">-25.790734</td>
<td align="right">0.0000000</td>
<td align="right">-24.730694</td>
</tr>
<tr class="odd">
<td>ENSG00000283460</td>
<td align="left">NA</td>
<td align="right">387.29128</td>
<td align="right">12.657692</td>
<td align="right">0.0000000</td>
<td align="right">-25.385382</td>
<td align="right">0.0000000</td>
<td align="right">-24.337678</td>
</tr>
<tr class="even">
<td>ENSG00000224472</td>
<td align="left">TCF19</td>
<td align="right">372.57228</td>
<td align="right">12.600198</td>
<td align="right">0.0000000</td>
<td align="right">-25.445591</td>
<td align="right">0.0000000</td>
<td align="right">-24.396110</td>
</tr>
<tr class="odd">
<td>ENSG00000262243</td>
<td align="left">CES1</td>
<td align="right">272.06221</td>
<td align="right">12.143572</td>
<td align="right">0.0000000</td>
<td align="right">-21.969075</td>
<td align="right">0.0000000</td>
<td align="right">-21.006050</td>
</tr>
<tr class="even">
<td>ENSG00000262327</td>
<td align="left">AGK</td>
<td align="right">253.20484</td>
<td align="right">12.042178</td>
<td align="right">0.0000000</td>
<td align="right">-15.844717</td>
<td align="right">0.0000000</td>
<td align="right">-15.041713</td>
</tr>
<tr class="odd">
<td>ENSG00000223654</td>
<td align="left">FLOT1</td>
<td align="right">210.34275</td>
<td align="right">11.775529</td>
<td align="right">0.0000000</td>
<td align="right">-22.301366</td>
<td align="right">0.0000000</td>
<td align="right">-21.331193</td>
</tr>
<tr class="even">
<td>ENSG00000282735</td>
<td align="left">KIAA0355</td>
<td align="right">175.86971</td>
<td align="right">11.518479</td>
<td align="right">0.0000000</td>
<td align="right">-14.421891</td>
<td align="right">0.0000000</td>
<td align="right">-13.658163</td>
</tr>
<tr class="odd">
<td>ENSG00000233890</td>
<td align="left">TCF19</td>
<td align="right">316.47938</td>
<td align="right">-11.363763</td>
<td align="right">0.0000000</td>
<td align="right">-27.297103</td>
<td align="right">0.0000000</td>
<td align="right">-26.197231</td>
</tr>
<tr class="even">
<td>ENSG00000281289</td>
<td align="left">LSS</td>
<td align="right">153.04846</td>
<td align="right">11.315626</td>
<td align="right">0.0000000</td>
<td align="right">-20.336677</td>
<td align="right">0.0000000</td>
<td align="right">-19.415177</td>
</tr>
<tr class="odd">
<td>ENSG00000230043</td>
<td align="left">NA</td>
<td align="right">609.14628</td>
<td align="right">11.293355</td>
<td align="right">0.0000000</td>
<td align="right">-26.941269</td>
<td align="right">0.0000000</td>
<td align="right">-25.851005</td>
</tr>
<tr class="even">
<td>ENSG00000224587</td>
<td align="left">MDC1</td>
<td align="right">297.25882</td>
<td align="right">-11.273461</td>
<td align="right">0.0000000</td>
<td align="right">-27.013565</td>
<td align="right">0.0000000</td>
<td align="right">-25.921341</td>
</tr>
<tr class="odd">
<td>ENSG00000173867</td>
<td align="left">NA</td>
<td align="right">131.49720</td>
<td align="right">11.098474</td>
<td align="right">0.0001189</td>
<td align="right">-3.924739</td>
<td align="right">0.0002957</td>
<td align="right">-3.529176</td>
</tr>
<tr class="even">
<td>ENSG00000259132</td>
<td align="left">NA</td>
<td align="right">127.78513</td>
<td align="right">11.055735</td>
<td align="right">0.0000000</td>
<td align="right">-18.763573</td>
<td align="right">0.0000000</td>
<td align="right">-17.882223</td>
</tr>
<tr class="odd">
<td>ENSG00000174028</td>
<td align="left">NA</td>
<td align="right">245.17011</td>
<td align="right">-10.995498</td>
<td align="right">0.0000000</td>
<td align="right">-25.743198</td>
<td align="right">0.0000000</td>
<td align="right">-24.685238</td>
</tr>
<tr class="even">
<td>ENSG00000257529</td>
<td align="left">RPL36A-HNRNPH2</td>
<td align="right">239.17911</td>
<td align="right">-10.959490</td>
<td align="right">0.0000000</td>
<td align="right">-25.340422</td>
<td align="right">0.0000000</td>
<td align="right">-24.294740</td>
</tr>
<tr class="odd">
<td>ENSG00000206492</td>
<td align="left">GNL1</td>
<td align="right">232.70451</td>
<td align="right">-10.920244</td>
<td align="right">0.0000000</td>
<td align="right">-25.262280</td>
<td align="right">0.0000000</td>
<td align="right">-24.218108</td>
</tr>
<tr class="even">
<td>ENSG00000262666</td>
<td align="left">FAM189B</td>
<td align="right">102.83438</td>
<td align="right">10.744264</td>
<td align="right">0.0000000</td>
<td align="right">-16.952593</td>
<td align="right">0.0000000</td>
<td align="right">-16.121128</td>
</tr>
<tr class="odd">
<td>ENSG00000223957</td>
<td align="left">NEU1</td>
<td align="right">100.73427</td>
<td align="right">10.713972</td>
<td align="right">0.0001557</td>
<td align="right">-3.807719</td>
<td align="right">0.0003826</td>
<td align="right">-3.417262</td>
</tr>
<tr class="even">
<td>ENSG00000243251</td>
<td align="left">NA</td>
<td align="right">97.08863</td>
<td align="right">10.659147</td>
<td align="right">0.0000000</td>
<td align="right">-17.957026</td>
<td align="right">0.0000000</td>
<td align="right">-17.099049</td>
</tr>
<tr class="odd">
<td>ENSG00000236428</td>
<td align="left">PPP1R18</td>
<td align="right">95.55938</td>
<td align="right">10.636517</td>
<td align="right">0.0000000</td>
<td align="right">-18.008242</td>
<td align="right">0.0000000</td>
<td align="right">-17.148460</td>
</tr>
<tr class="even">
<td>ENSG00000182774</td>
<td align="left">RPS17</td>
<td align="right">187.07199</td>
<td align="right">-10.605096</td>
<td align="right">0.0000000</td>
<td align="right">-23.859227</td>
<td align="right">0.0000000</td>
<td align="right">-22.854892</td>
</tr>
<tr class="odd">
<td>ENSG00000282785</td>
<td align="left">NA</td>
<td align="right">1001.73943</td>
<td align="right">10.546736</td>
<td align="right">0.0000000</td>
<td align="right">-66.191441</td>
<td align="right">0.0000000</td>
<td align="right">-64.394026</td>
</tr>
<tr class="even">
<td>ENSG00000228760</td>
<td align="left">BAG6</td>
<td align="right">159.51923</td>
<td align="right">-10.375144</td>
<td align="right">0.0000000</td>
<td align="right">-21.825535</td>
<td align="right">0.0000000</td>
<td align="right">-20.865418</td>
</tr>
<tr class="odd">
<td>ENSG00000283243</td>
<td align="left">SHANK3</td>
<td align="right">149.52376</td>
<td align="right">-10.281902</td>
<td align="right">0.0000000</td>
<td align="right">-22.369858</td>
<td align="right">0.0000000</td>
<td align="right">-21.397347</td>
</tr>
<tr class="even">
<td>ENSG00000281948</td>
<td align="left">LOC107987423</td>
<td align="right">70.89664</td>
<td align="right">10.205355</td>
<td align="right">0.0000000</td>
<td align="right">-16.219319</td>
<td align="right">0.0000000</td>
<td align="right">-15.406690</td>
</tr>
<tr class="odd">
<td>ENSG00000253945</td>
<td align="left">NA</td>
<td align="right">67.11174</td>
<td align="right">10.128958</td>
<td align="right">0.0000000</td>
<td align="right">-14.367054</td>
<td align="right">0.0000000</td>
<td align="right">-13.604773</td>
</tr>
<tr class="even">
<td>ENSG00000283553</td>
<td align="left">NA</td>
<td align="right">65.56403</td>
<td align="right">10.093708</td>
<td align="right">0.0000000</td>
<td align="right">-15.632485</td>
<td align="right">0.0000000</td>
<td align="right">-14.836633</td>
</tr>
<tr class="odd">
<td>ENSG00000234196</td>
<td align="left">ZBTB12</td>
<td align="right">130.40189</td>
<td align="right">-10.084548</td>
<td align="right">0.0000000</td>
<td align="right">-20.816319</td>
<td align="right">0.0000000</td>
<td align="right">-19.882324</td>
</tr>
<tr class="even">
<td>ENSG00000263020</td>
<td align="left">NA</td>
<td align="right">63.82464</td>
<td align="right">10.062020</td>
<td align="right">0.0000000</td>
<td align="right">-13.527041</td>
<td align="right">0.0000000</td>
<td align="right">-12.788400</td>
</tr>
<tr class="odd">
<td>ENSG00000280852</td>
<td align="left">LOC653653</td>
<td align="right">249.37149</td>
<td align="right">10.029893</td>
<td align="right">0.0000000</td>
<td align="right">-39.067065</td>
<td align="right">0.0000000</td>
<td align="right">-37.730819</td>
</tr>
<tr class="even">
<td>ENSG00000203812</td>
<td align="left">HIST2H2AA3</td>
<td align="right">59.38863</td>
<td align="right">9.950450</td>
<td align="right">0.0000000</td>
<td align="right">-15.155185</td>
<td align="right">0.0000000</td>
<td align="right">-14.375487</td>
</tr>
<tr class="odd">
<td>ENSG00000234648</td>
<td align="left">NA</td>
<td align="right">115.78445</td>
<td align="right">-9.913479</td>
<td align="right">0.0000000</td>
<td align="right">-20.235413</td>
<td align="right">0.0000000</td>
<td align="right">-19.316180</td>
</tr>
<tr class="even">
<td>ENSG00000234964</td>
<td align="left">NA</td>
<td align="right">106.16769</td>
<td align="right">-9.787543</td>
<td align="right">0.0000000</td>
<td align="right">-18.517243</td>
<td align="right">0.0000000</td>
<td align="right">-17.642408</td>
</tr>
<tr class="odd">
<td>ENSG00000275300</td>
<td align="left">MCCC2</td>
<td align="right">82.00449</td>
<td align="right">9.779681</td>
<td align="right">0.0000000</td>
<td align="right">-14.223894</td>
<td align="right">0.0000000</td>
<td align="right">-13.466835</td>
</tr>
<tr class="even">
<td>ENSG00000277641</td>
<td align="left">WNT3</td>
<td align="right">100.41017</td>
<td align="right">-9.707385</td>
<td align="right">0.0075049</td>
<td align="right">-2.124653</td>
<td align="right">0.0146514</td>
<td align="right">-1.834120</td>
</tr>
<tr class="odd">
<td>ENSG00000236418</td>
<td align="left">HLA-DQA1</td>
<td align="right">100.16776</td>
<td align="right">-9.703914</td>
<td align="right">0.0075264</td>
<td align="right">-2.123415</td>
<td align="right">0.0146888</td>
<td align="right">-1.833014</td>
</tr>
<tr class="even">
<td>ENSG00000223618</td>
<td align="left">VPS52</td>
<td align="right">48.40785</td>
<td align="right">9.656625</td>
<td align="right">0.0003463</td>
<td align="right">-3.460597</td>
<td align="right">0.0008188</td>
<td align="right">-3.086830</td>
</tr>
<tr class="odd">
<td>ENSG00000267059</td>
<td align="left">NA</td>
<td align="right">254.83601</td>
<td align="right">-9.617788</td>
<td align="right">0.0000000</td>
<td align="right">-20.436333</td>
<td align="right">0.0000000</td>
<td align="right">-19.513124</td>
</tr>
<tr class="even">
<td>ENSG00000244563</td>
<td align="left">NA</td>
<td align="right">40.14673</td>
<td align="right">9.385623</td>
<td align="right">0.0000000</td>
<td align="right">-12.715493</td>
<td align="right">0.0000000</td>
<td align="right">-11.996903</td>
</tr>
<tr class="odd">
<td>ENSG00000114790</td>
<td align="left">ARHGEF26</td>
<td align="right">80.13651</td>
<td align="right">-9.382221</td>
<td align="right">0.0000000</td>
<td align="right">-17.757058</td>
<td align="right">0.0000000</td>
<td align="right">-16.905580</td>
</tr>
<tr class="even">
<td>ENSG00000258677</td>
<td align="left">NA</td>
<td align="right">76.59470</td>
<td align="right">-9.317170</td>
<td align="right">0.0000000</td>
<td align="right">-18.036393</td>
<td align="right">0.0000000</td>
<td align="right">-17.175624</td>
</tr>
<tr class="odd">
<td>ENSG00000124205</td>
<td align="left">EDN3</td>
<td align="right">268.99825</td>
<td align="right">9.313617</td>
<td align="right">0.0000000</td>
<td align="right">-31.109931</td>
<td align="right">0.0000000</td>
<td align="right">-29.927551</td>
</tr>
<tr class="even">
<td>ENSG00000267534</td>
<td align="left">S1PR2</td>
<td align="right">37.28580</td>
<td align="right">9.280728</td>
<td align="right">0.0000000</td>
<td align="right">-13.338519</td>
<td align="right">0.0000000</td>
<td align="right">-12.605930</td>
</tr>
<tr class="odd">
<td>ENSG00000237580</td>
<td align="left">NA</td>
<td align="right">36.99971</td>
<td align="right">9.269904</td>
<td align="right">0.0000000</td>
<td align="right">-12.009517</td>
<td align="right">0.0000000</td>
<td align="right">-11.313263</td>
</tr>
<tr class="even">
<td>ENSG00000273398</td>
<td align="left">NA</td>
<td align="right">72.06234</td>
<td align="right">-9.228464</td>
<td align="right">0.0000000</td>
<td align="right">-16.994242</td>
<td align="right">0.0000000</td>
<td align="right">-16.161542</td>
</tr>
<tr class="odd">
<td>ENSG00000281742</td>
<td align="left">MCCC2</td>
<td align="right">66.61644</td>
<td align="right">-9.115579</td>
<td align="right">0.0000000</td>
<td align="right">-16.854846</td>
<td align="right">0.0000000</td>
<td align="right">-16.026143</td>
</tr>
<tr class="even">
<td>ENSG00000166783</td>
<td align="left">MARF1</td>
<td align="right">65.98720</td>
<td align="right">-9.101793</td>
<td align="right">0.0000000</td>
<td align="right">-14.286921</td>
<td align="right">0.0000000</td>
<td align="right">-13.527128</td>
</tr>
<tr class="odd">
<td>ENSG00000272780</td>
<td align="left">NA</td>
<td align="right">63.89816</td>
<td align="right">-9.055748</td>
<td align="right">0.0000000</td>
<td align="right">-16.919849</td>
<td align="right">0.0000000</td>
<td align="right">-16.088691</td>
</tr>
<tr class="even">
<td>ENSG00000278573</td>
<td align="left">NA</td>
<td align="right">31.76068</td>
<td align="right">9.048547</td>
<td align="right">0.0006120</td>
<td align="right">-3.213241</td>
<td align="right">0.0013989</td>
<td align="right">-2.854199</td>
</tr>
<tr class="odd">
<td>ENSG00000283149</td>
<td align="left">NA</td>
<td align="right">59.69974</td>
<td align="right">-8.958003</td>
<td align="right">0.0000000</td>
<td align="right">-16.044382</td>
<td align="right">0.0000000</td>
<td align="right">-15.236883</td>
</tr>
<tr class="even">
<td>ENSG00000262633</td>
<td align="left">NA</td>
<td align="right">56.52239</td>
<td align="right">-8.878874</td>
<td align="right">0.0000000</td>
<td align="right">-16.128320</td>
<td align="right">0.0000000</td>
<td align="right">-15.318190</td>
</tr>
<tr class="odd">
<td>ENSG00000227317</td>
<td align="left">DDAH2</td>
<td align="right">55.09922</td>
<td align="right">-8.841597</td>
<td align="right">0.0148923</td>
<td align="right">-1.827037</td>
<td align="right">0.0276110</td>
<td align="right">-1.558917</td>
</tr>
<tr class="even">
<td>ENSG00000152086</td>
<td align="left">TUBA3E</td>
<td align="right">27.38916</td>
<td align="right">8.835309</td>
<td align="right">0.0000000</td>
<td align="right">-11.428499</td>
<td align="right">0.0000000</td>
<td align="right">-10.748410</td>
</tr>
<tr class="odd">
<td>ENSG00000256968</td>
<td align="left">NA</td>
<td align="right">26.74882</td>
<td align="right">8.801180</td>
<td align="right">0.0000000</td>
<td align="right">-10.765160</td>
<td align="right">0.0000000</td>
<td align="right">-10.108654</td>
</tr>
<tr class="even">
<td>ENSG00000267697</td>
<td align="left">LUZP6</td>
<td align="right">26.26094</td>
<td align="right">8.774552</td>
<td align="right">0.0005166</td>
<td align="right">-3.286876</td>
<td align="right">0.0011933</td>
<td align="right">-2.923244</td>
</tr>
<tr class="odd">
<td>ENSG00000235692</td>
<td align="left">PFDN6</td>
<td align="right">50.95973</td>
<td align="right">-8.728953</td>
<td align="right">0.0162194</td>
<td align="right">-1.789966</td>
<td align="right">0.0298458</td>
<td align="right">-1.525117</td>
</tr>
<tr class="even">
<td>ENSG00000231878</td>
<td align="left">NA</td>
<td align="right">25.29156</td>
<td align="right">8.721717</td>
<td align="right">0.0000000</td>
<td align="right">-10.994723</td>
<td align="right">0.0000000</td>
<td align="right">-10.328010</td>
</tr>
<tr class="odd">
<td>ENSG00000282684</td>
<td align="left">ZC3H3</td>
<td align="right">47.88194</td>
<td align="right">-8.639698</td>
<td align="right">0.0000000</td>
<td align="right">-14.959121</td>
<td align="right">0.0000000</td>
<td align="right">-14.184857</td>
</tr>
<tr class="even">
<td>ENSG00000167711</td>
<td align="left">SERPINF2</td>
<td align="right">45.21632</td>
<td align="right">-8.557066</td>
<td align="right">0.0000000</td>
<td align="right">-11.932473</td>
<td align="right">0.0000000</td>
<td align="right">-11.239254</td>
</tr>
<tr class="odd">
<td>ENSG00000280158</td>
<td align="left">NA</td>
<td align="right">45.15382</td>
<td align="right">-8.554864</td>
<td align="right">0.0000000</td>
<td align="right">-14.299419</td>
<td align="right">0.0000000</td>
<td align="right">-13.538842</td>
</tr>
<tr class="even">
<td>ENSG00000242361</td>
<td align="left">HLA-DMA</td>
<td align="right">21.74032</td>
<td align="right">8.499219</td>
<td align="right">0.0000000</td>
<td align="right">-9.344540</td>
<td align="right">0.0000000</td>
<td align="right">-8.733071</td>
</tr>
<tr class="odd">
<td>ENSG00000186076</td>
<td align="left">NA</td>
<td align="right">565.21314</td>
<td align="right">8.420632</td>
<td align="right">0.0000973</td>
<td align="right">-4.011969</td>
<td align="right">0.0002439</td>
<td align="right">-3.612836</td>
</tr>
<tr class="even">
<td>ENSG00000273707</td>
<td align="left">CDKN1C</td>
<td align="right">151.56088</td>
<td align="right">8.327926</td>
<td align="right">0.0000227</td>
<td align="right">-4.643106</td>
<td align="right">0.0000611</td>
<td align="right">-4.213688</td>
</tr>
<tr class="odd">
<td>ENSG00000185641</td>
<td align="left">NA</td>
<td align="right">190.70288</td>
<td align="right">8.313943</td>
<td align="right">0.0000000</td>
<td align="right">-12.647322</td>
<td align="right">0.0000000</td>
<td align="right">-11.929681</td>
</tr>
<tr class="even">
<td>ENSG00000230336</td>
<td align="left">POU5F1</td>
<td align="right">18.56966</td>
<td align="right">8.274299</td>
<td align="right">0.0246028</td>
<td align="right">-1.609016</td>
<td align="right">0.0437611</td>
<td align="right">-1.358912</td>
</tr>
<tr class="odd">
<td>ENSG00000277086</td>
<td align="left">MTMR10</td>
<td align="right">34.41780</td>
<td align="right">-8.162631</td>
<td align="right">0.0000000</td>
<td align="right">-12.826292</td>
<td align="right">0.0000000</td>
<td align="right">-12.105678</td>
</tr>
<tr class="even">
<td>ENSG00000263809</td>
<td align="left">NA</td>
<td align="right">33.67410</td>
<td align="right">-8.131452</td>
<td align="right">0.0000000</td>
<td align="right">-12.245246</td>
<td align="right">0.0000000</td>
<td align="right">-11.541252</td>
</tr>
<tr class="odd">
<td>ENSG00000230143</td>
<td align="left">FLOT1</td>
<td align="right">33.62465</td>
<td align="right">-8.129907</td>
<td align="right">0.0000000</td>
<td align="right">-12.652086</td>
<td align="right">0.0000000</td>
<td align="right">-11.934208</td>
</tr>
<tr class="even">
<td>ENSG00000180581</td>
<td align="left">NA</td>
<td align="right">33.31416</td>
<td align="right">-8.115694</td>
<td align="right">0.0254239</td>
<td align="right">-1.594758</td>
<td align="right">0.0451042</td>
<td align="right">-1.345783</td>
</tr>
<tr class="odd">
<td>ENSG00000263620</td>
<td align="left">NA</td>
<td align="right">32.91314</td>
<td align="right">-8.098944</td>
<td align="right">0.0000000</td>
<td align="right">-12.897451</td>
<td align="right">0.0000000</td>
<td align="right">-12.174565</td>
</tr>
<tr class="even">
<td>ENSG00000267001</td>
<td align="left">NA</td>
<td align="right">32.14571</td>
<td align="right">-8.064379</td>
<td align="right">0.0000000</td>
<td align="right">-12.932927</td>
<td align="right">0.0000000</td>
<td align="right">-12.209080</td>
</tr>
<tr class="odd">
<td>ENSG00000280571</td>
<td align="left">NA</td>
<td align="right">52.23861</td>
<td align="right">-8.052881</td>
<td align="right">0.0000000</td>
<td align="right">-11.591445</td>
<td align="right">0.0000000</td>
<td align="right">-10.907535</td>
</tr>
<tr class="even">
<td>ENSG00000253816</td>
<td align="left">NA</td>
<td align="right">31.31488</td>
<td align="right">-8.025978</td>
<td align="right">0.0000000</td>
<td align="right">-12.587029</td>
<td align="right">0.0000000</td>
<td align="right">-11.871162</td>
</tr>
<tr class="odd">
<td>ENSG00000217733</td>
<td align="left">NA</td>
<td align="right">84.65976</td>
<td align="right">7.998947</td>
<td align="right">0.0000000</td>
<td align="right">-17.195517</td>
<td align="right">0.0000000</td>
<td align="right">-16.356595</td>
</tr>
<tr class="even">
<td>ENSG00000277044</td>
<td align="left">OPRL1</td>
<td align="right">15.28371</td>
<td align="right">7.992178</td>
<td align="right">0.0013284</td>
<td align="right">-2.876658</td>
<td align="right">0.0029110</td>
<td align="right">-2.535957</td>
</tr>
<tr class="odd">
<td>ENSG00000277278</td>
<td align="left">HERC2</td>
<td align="right">29.00972</td>
<td align="right">-7.916123</td>
<td align="right">0.0000000</td>
<td align="right">-12.342978</td>
<td align="right">0.0000000</td>
<td align="right">-11.634831</td>
</tr>
<tr class="even">
<td>ENSG00000270136</td>
<td align="left">MINOS1-NBL1</td>
<td align="right">179.64712</td>
<td align="right">-7.913475</td>
<td align="right">0.0000000</td>
<td align="right">-10.336730</td>
<td align="right">0.0000000</td>
<td align="right">-9.694401</td>
</tr>
<tr class="odd">
<td>ENSG00000232569</td>
<td align="left">GABBR1</td>
<td align="right">13.98351</td>
<td align="right">7.863862</td>
<td align="right">0.0000000</td>
<td align="right">-8.680086</td>
<td align="right">0.0000000</td>
<td align="right">-8.089629</td>
</tr>
<tr class="even">
<td>ENSG00000275326</td>
<td align="left">NOSTRIN</td>
<td align="right">30.40018</td>
<td align="right">7.826545</td>
<td align="right">0.0000001</td>
<td align="right">-6.996009</td>
<td align="right">0.0000003</td>
<td align="right">-6.471235</td>
</tr>
<tr class="odd">
<td>ENSG00000277150</td>
<td align="left">F8A3</td>
<td align="right">26.93577</td>
<td align="right">-7.809527</td>
<td align="right">0.0000000</td>
<td align="right">-11.650948</td>
<td align="right">0.0000000</td>
<td align="right">-10.965390</td>
</tr>
<tr class="even">
<td>ENSG00000171532</td>
<td align="left">NEUROD2</td>
<td align="right">244.57296</td>
<td align="right">7.796966</td>
<td align="right">0.0000000</td>
<td align="right">-41.285240</td>
<td align="right">0.0000000</td>
<td align="right">-39.916278</td>
</tr>
<tr class="odd">
<td>ENSG00000136244</td>
<td align="left">IL6</td>
<td align="right">25.65669</td>
<td align="right">-7.734676</td>
<td align="right">0.0000000</td>
<td align="right">-10.470953</td>
<td align="right">0.0000000</td>
<td align="right">-9.824219</td>
</tr>
<tr class="even">
<td>ENSG00000227222</td>
<td align="left">DHX16</td>
<td align="right">24.77538</td>
<td align="right">-7.689310</td>
<td align="right">0.0000000</td>
<td align="right">-11.076587</td>
<td align="right">0.0000000</td>
<td align="right">-10.407232</td>
</tr>
<tr class="odd">
<td>ENSG00000273003</td>
<td align="left">ARL2-SNX15</td>
<td align="right">24.74304</td>
<td align="right">-7.686421</td>
<td align="right">0.0000000</td>
<td align="right">-10.523655</td>
<td align="right">0.0000000</td>
<td align="right">-9.875005</td>
</tr>
<tr class="even">
<td>ENSG00000227402</td>
<td align="left">SLC39A7</td>
<td align="right">88.92664</td>
<td align="right">7.680142</td>
<td align="right">0.0084928</td>
<td align="right">-2.070952</td>
<td align="right">0.0164157</td>
<td align="right">-1.784740</td>
</tr>
<tr class="odd">
<td>ENSG00000120675</td>
<td align="left">DNAJC15</td>
<td align="right">135.07129</td>
<td align="right">7.627673</td>
<td align="right">0.0000000</td>
<td align="right">-39.815275</td>
<td align="right">0.0000000</td>
<td align="right">-38.465018</td>
</tr>
<tr class="even">
<td>ENSG00000234218</td>
<td align="left">MICB</td>
<td align="right">23.05480</td>
<td align="right">-7.585511</td>
<td align="right">0.0000000</td>
<td align="right">-10.570884</td>
<td align="right">0.0000000</td>
<td align="right">-9.920917</td>
</tr>
</tbody>
</table>
</div>
<div id="gene-clustering" class="section level2">
<h2>Gene clustering</h2>
<div id="clustering-of-all-selected-genes-by-expression-level-rlog-transformed" class="section level3">
<h3>Clustering of all selected genes by expression level (rlog transformed)</h3>
<pre class="r"><code>library(genefilter)</code></pre>
<pre><code>## 
## Attaching package: &#39;genefilter&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     rowSds, rowVars</code></pre>
<pre class="r"><code>mat &lt;- assay(rld)[rownames(selected), ]
anno &lt;- as.data.frame(colData(rld)[, c(&quot;condition&quot;, &quot;sample_name&quot;)])
pheatmap(mat, annotation_col = anno, show_rownames = FALSE,
         main = &quot;Heatmap of rlog-transformed gene expression levels&quot;)</code></pre>
<p><img src="rnaseq_explore_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
</div>
<div id="clustering-of-all-selected-genes-by-relative-expression-levels-across-samples" class="section level3">
<h3>Clustering of all selected genes by relative expression levels across samples</h3>
<p>Instead of absolute expression strength, the following plot uses the amount by which each gene deviates in a specific sample from the gene’s average across all samples.</p>
<p>The trees on top shows the clustering by samples, the trees on left shows the clustering by genes. It can be shown that the control and treat samples are clearly separated and the genes may be roughly divided into two clusters (up-regulated and down-regulated).</p>
<pre class="r"><code>mat &lt;- mat - rowMeans(mat)
pheatmap(mat, annotation_col = anno, show_rownames = FALSE,
         main = &quot;Heatmap of relative rlog-transformed values across samples&quot;)</code></pre>
<p><img src="rnaseq_explore_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
</div>
<div id="clustering-of-genes-with-the-highest-variance-across-samples" class="section level3">
<h3>Clustering of genes with the highest variance across samples</h3>
<p>Here we select the 30 genes with the highest variance of expression values across samples.</p>
<pre class="r"><code>library(genefilter)
topVarGenes &lt;- head(order(rowVars(assay(rld)), decreasing = TRUE), 30)
mat &lt;- assay(rld)[topVarGenes, ]
mat &lt;- mat[rownames(mat) %in% rownames(selected), ]
mat &lt;- mat - rowMeans(mat)
anno &lt;- as.data.frame(colData(rld)[, c(&quot;condition&quot;, &quot;sample_name&quot;)])
pheatmap(mat, annotation_col = anno,
         main = &quot;Heatmap of relative rlog-transformed values across samples&quot;)</code></pre>
<p><img src="rnaseq_explore_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
</div>
</div>
<div id="pca-plot-of-genes" class="section level2">
<h2>PCA plot of genes</h2>
<p>We will use rlog-transformed values. The following is a PCA plot of all selected genes.</p>
<pre class="r"><code>library(ggfortify)
autoplot(prcomp(assay(rld)[rownames(assay(rld)) %in% rownames(selected), ]), size = 1, alpha = 0.8)</code></pre>
<p><img src="rnaseq_explore_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<p>It is clear that the genes can be separated by two parts by the second dimension. If we plot all the genes like the following, it will be clear for us that the second dimension aligns with the foldchange – since we have filtered the genes with low foldchange. Thus the two clusters are up-regulated genes and down-regulated genes.</p>
<pre class="r"><code>autoplot(prcomp(assay(rld)[rownames(assay(rld)), ]), size = 1, alpha = 0.8)</code></pre>
<p><img src="rnaseq_explore_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
<div id="pca-plot-of-genes-with-high-variance" class="section level3">
<h3>PCA plot of genes with high variance</h3>
<p>Again, we select the genes with the highest variance across samples:</p>
<pre class="r"><code>topVarGenes &lt;- head(order(rowVars(assay(rld)), decreasing = TRUE), 50)
tmp &lt;- assay(rld)[topVarGenes, ]
tmp &lt;- tmp[rownames(tmp) %in% rownames(selected), ]
autoplot(prcomp(tmp))</code></pre>
<p><img src="rnaseq_explore_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
